cmd/cmd.go

```diff
@@ -112,32 +115,30 @@ func coreSupportsReExec(corePath string) bool {
 func InternalToolPath(tool string) string {
 	distroTool := filepath.Join(dirs.DistroLibExecDir, tool)
 
-	// If we are asked not to re-execute use distribution packages. This is
-	// "spiritual" re-exec so use the same environment variable to decide.
+	// mostly useful for tests
 	if !osutil.GetenvBool(reExecKey, true) {
 		logger.Debugf("re-exec disabled by user")
 		return distroTool
 	}
 
-	// If the distribution doesn't support re-exec or run-from-core then don't do it.
-	if !distroSupportsReExec() {
+	exe, err := osReadlink("/proc/self/exe")
```

Comment:
Can we please have a reminder here explaining why we're not just using the straight path, to make sure we won't revert into the bug by mistake?